hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+
+# Theory of Operations
+
+## SPI_HOST IP Command Interface
+
+Issuing a command generally consists of the following steps:
+1. Configure the IP to be compatible with the target peripheral, using the {{< regref "CONFIG" >}} register,
+2. Load the instructions and data to be transmitted into the TXFIFO via the {{< regref "TXDATA" >}} register,
+3. Issue a 32-bit command to the {{< regref "COMMAND" >}} register with the following information,
+    - The speed to use for this particular command (Standard, Dual, or Quad)
+    - The three bit chip-select ID of the peripheral to target
+    - The number of bytes to transmit at *reduced* data rate, using `sd[0]` only (Dual or Quad rate *only*)
+        - This field is specifically for Dual or Quad commands where the first instruction must be sent serially on `sd[0]`
+    - The number of bytes to transmit at full data rate
+        - In Standard mode this controls the number of bytes transmitted and the number of bytes received, since the bus is full-duplex in Standard mode.
+    - The number of dummy `sck` cycles to wait before reading data (Dual or Quad rate *only*)
+    - The number of bytes to receive at the full Dual or Quad rate (Dual or Quad rate *only*)
+4. and read the peripheral response from the {{< regref "RXDATA" >}} register.
+
+### Configuration options
+
+In general, the {{< regref "CONFIG" >}} register holds clock configuration and timing settings which are specific to a particular class of target device.
+This register should be updated whenever a target peripheral requires different timing parameters than the device selected in the previous command.
+If all of the devices attached to the SPI_HOST have the same clock polarity, phase and timing requirements, the {{< regref "CONFIG" >}} register can in principle be initialized once and left unchanged for all subsequent commands.
+
+One of the most common differences between target devices is the requirement for a specific SPI clock phase or polarity, CPOL and CPHA, which were described in the previous section [SPI Protocol Basics](#spi-protocol-basics).
+These clock parameters can be set via the {{< regref "CONFIG.CPOL">}} or {{< regref "CONFIG.CPHA" >}} register fields.
+Likewise, as also described in the previous section, if device setup times require a full clock cycle before sampling the output, Full-Cycle Mode can be enabled by asserting the {{< regref "CONFIG.FULLCYC" >}} bit.
+
+#### Clock rate selection
+
+The SPI clock rate for each peripheral is set by two factors:
+- The SPI_HOST core clock (which may be generated independently from the TL-UL iterface clock)
+- An additional 16-bit clock divider
+
+The SPI protocol usually requires activity (either sampling or asserting data), and either edge of the `sck` clock.
+For this reason the maximum `sck` frequency is at most one half the SPI_HOST core frequency.
```

Comment:
Ah okay noted.  Do you think there's an easy to add support for div_by_1 clock? Or would you prefer the SoC side supply 2x the clock?

Reply:
I think the SOC side should provide 2x of whatever clock they need.  I think this is pretty typical for these blocks.  I don't think PD would like it if we randomly had an FSM operating on both edges of the input clock.   The code would also look like a disaster, especially when trying to support multiple phases or polarities.

Reply:
sounds good. If it's not already, would you mind listing this very high in the feature list? 
That way we make the integrator (in this case me...) won't miss this and will provision a 2x clock. 

It's actually good that you're making spi clock separate frequency from the bus clock.. since most likely i'm going to make the latter faster and the former a bit slower.

Reply:
Added to feature list:
- SPI clock rate controlled by separate input clock to core
   - SPI `sck` line typically toggles at 1/2 the core clock frequency.
   - An additional clock rate divider exists to reduce the frequency if needed